core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushLimitThroughMarkDistinct.java

```diff
@@ -33,6 +34,7 @@
 
     // Applies to both limit with ties and limit without ties.
     private static final Pattern<LimitNode> PATTERN = limit()
+            .with(isOrderSensitive().equalTo(false))
```

Comment:
MarkDistinct doesn't change the relative order of the input, so this should be allowed as long as the ordering scheme does not reference the marker column

Reply:
Making this change.
Should there be a PushTopNThroughMarkDistinct rule as well ?